exercises/luhn/src/example.c

```diff
@@ -0,0 +1,38 @@
+#include "luhn.h"
+#include <stddef.h>
+#include <string.h>
+#include <ctype.h>
+
+bool luhn(const char *num)
+{
+   if (num == NULL)
+      return false;
+
+   int pos = strlen(num) - 1;
+   if (pos < 0)
+      return false;
+
+   int res = 0;
+   int every_second = 0;
+   int zeros = 0;
+   while (pos >= 0) {
+      if (isdigit(num[pos])) {
+         char c = num[pos] - '0';
+         zeros += (c == 0);
+         if (every_second) {
+            c *= 2;
+            res += c > 9 ? c - 9 : c;
+         } else {
+            res += c;
+         }
+         every_second = !every_second;
+      } else if (num[pos] != ' ') {
```

Comment:
The readme states that spaces are allowed in input (but that they must be stripped before checking). Further, as the nature of allowed space is not specifically defined (to be 0x32), it may be better to use `isspace()` if checking for it.

Reply:
@wolf99 I think that you misread this as `==` instead of `!=`. This is stating that if it's not a digit _and it's not a space_ then it's invalid.

Reply:
@ryanplusplus You are correct, I read that incorrectly 🙄